tests/MongoDB.Driver.Core.Tests/Specifications/read-write-concern/ConnectionStringTestRunner.cs

```diff
@@ -76,13 +72,28 @@ private void AssertValid(ConnectionString connectionString, BsonDocument definit
             BsonValue writeConcernValue;
             if (definition.TryGetValue("writeConcern", out writeConcernValue))
             {
-                var writeConcern = WriteConcern.FromBsonDocument(MassageWriteConcernDocument((BsonDocument)writeConcernValue));
+                try
+                {
+                    var writeConcern =
+                        WriteConcern.FromBsonDocument(MassageWriteConcernDocument((BsonDocument) writeConcernValue));
+                    connectionString.W.Should().Be(writeConcern.W);
+                    connectionString.WTimeout.Should().Be(writeConcern.WTimeout);
+                    connectionString.Journal.Should().Be(writeConcern.Journal);
+                    connectionString.FSync.Should().Be(writeConcern.FSync);
+                }
+                //if invalid parameter combination
+                catch (Exception ex)
+                {
+                    AssertInvalid(ex, definition);
+                    return;
+                }
+            }
 
-                connectionString.W.Should().Be(writeConcern.W);
-                connectionString.WTimeout.Should().Be(writeConcern.WTimeout);
-                connectionString.Journal.Should().Be(writeConcern.Journal);
-                connectionString.FSync.Should().Be(writeConcern.FSync);
+            if (!definition["valid"].ToBoolean())
+            {
+                throw new AssertionException($"The connection string '{definition["uri"]}' should be invalid.");
             }
+
```

Comment:
It seems wrong that we even called a method called `AssertValid` if we weren't expecting the write concern to be valid in the first place.

Maybe this test needs to be in the caller.

Reply:
Agreed, see above.